File: frontend/src/dtos/state-dto.ts

export class StateDTO {
    latitude!: number
    longitude!: number
    odometer!: number
    is_driving!: Boolean
    is_charging!: Boolean
    shift_state?: string
    battery_level!: number
    heading?: number
    speed?: number

    active_route_destination?: string
    active_route_latitude?: number
    active_route_longitude?: number
    active_route_minutes_to_arrival?: number
    active_route_energy_at_arrival?: number
}

File: dtos/state_dto.py
from dataclasses import field
from typing import Optional
from marshmallow_dataclass import dataclass

@dataclass
class StateDTO():
    latitude: float = None
    longitude: float = None
    heading: float = None
    odometer: float = None
    is_driving: bool = False
    is_charging: bool = False
    battery_level: int = 0

    active_route_destination: str = None
    active_route_latitude: float = None
    active_route_longitude: float = None
    active_route_minutes_to_arrival: float = None
    active_route_energy_at_arrival: int = None
